src/java/htsjdk/samtools/BamIndexValidator.java

```diff
@@ -84,4 +84,72 @@ public static int exhaustivelyTestIndex(final SamReader reader) { // throws Exce
         return 0;
     }
 
+    /**
+     * A less time-consuming index validation that only looks at the first and last references in the index
+     *
+     * @param reader
+     * @return # of chunks examined, or 9 if there is no browseable index for the reader
+     */
+    public static int lessExhaustivelyTestIndex(final SamReader reader) {
+        // look at all chunk offsets in a linear index to make sure they are valid
+
+        if (reader.indexing().hasBrowseableIndex()) {
+
+            // content is from an existing bai file
+            final CachingBAMFileIndex existingIndex = (CachingBAMFileIndex) reader.indexing().getBrowseableIndex();
+            final int n_ref = existingIndex.getNumberOfReferences();
+
+
+            int chunkCount = 0;
+            int indexCount = 0;
+            for (int i = 0; i < n_ref; i++) {
+
+                // Whether we are validating this reference
+                final boolean validateThisReference = (i == 0 || i == (n_ref-1));
+
+                final BAMIndexContent content = existingIndex.getQueryResults(i);
+
+                if (validateThisReference) {
+                    for (final Chunk c : content.getAllChunks()) {
+                        final CloseableIterator<SAMRecord> iter = ((SamReader.PrimitiveSamReaderToSamReaderAdapter) reader).iterator(new BAMFileSpan(c));
+                        chunkCount++;
+                        BAMRecord b = null;
+                        try {
+                            b = (BAMRecord) iter.next();
+                            iter.close();
+                        } catch (final Exception e) {
+                            throw new SAMException("Exception in BamIndexValidator. Last good record " + b + " in chunk " + c + " chunkCount=" + chunkCount, e);
+                        }
+                    }
+                }
+                else {
+                   chunkCount += content.getAllChunks().size();
+                }
+
+                // also seek to every position in the linear index
+                final LinearIndex linearIndex = content.getLinearIndex();
+                if (validateThisReference) {
+                    for (final long l : linearIndex.getIndexEntries()) {
+                        try {
+                            if (l != 0) {
+                                final CloseableIterator<SAMRecord> iter = ((SamReader.PrimitiveSamReaderToSamReaderAdapter) reader).iterator(new BAMFileSpan(new Chunk(l, l + 1)));

```

Comment:
@tfenne do you have any insight into why we need the linear index check here?  Can't we just remove this entire block of code?  At the very least, just check the first record for the contig and that's it?